pkg/pool-weighted/test/InvestmentPool.test.ts

```diff
@@ -124,4 +75,384 @@ describe('InvestmentPool', function () {
+            sender = owner;
+            await pool.init({ from: owner, initialBalances });
+          });
+
+          it('swaps can be enabled and disabled', async () => {
+            await pool.setSwapEnabled(sender, false);
+            expect(await pool.instance.getSwapEnabled()).to.be.false;
+
+            await pool.setSwapEnabled(sender, true);
+            expect(await pool.instance.getSwapEnabled()).to.be.true;
+          });
+
+          it('disabling swaps emits an event', async () => {
+            const receipt = await pool.setSwapEnabled(sender, false);
+
+            expectEvent.inReceipt(await receipt.wait(), 'SwapEnabledSet', {
+              swapEnabled: false,
+            });
+          });
+
+          it('enabling swaps emits an event', async () => {
+            const receipt = await pool.setSwapEnabled(sender, true);
+
+            expectEvent.inReceipt(await receipt.wait(), 'SwapEnabledSet', {
+              swapEnabled: true,
+            });
+          });
+
+          it('owner can join and receive BPT, then exit', async () => {
+            const bptBeforeJoin = await pool.balanceOf(owner.address);
+            await expect(pool.joinGivenIn({ from: owner, amountsIn: initialBalances })).to.not.be.reverted;
+
+            const bptAfterJoin = await pool.balanceOf(owner.address);
+            expect(bptAfterJoin).to.gt(bptBeforeJoin);
+
+            await expect(pool.exitGivenOut({ from: owner, amountsOut: initialBalances })).to.not.be.reverted;
+            const bptAfterExit = await pool.balanceOf(owner.address);
+            expect(bptAfterExit).to.lt(bptAfterJoin);
+          });
+
+          context('when swaps disabled', () => {
+            sharedBeforeEach(async () => {
+              await pool.setSwapEnabled(sender, false);
+            });
+
+            /* future
+            it('disallows disproportionate joins (single token)', async () => {
+              const bptOut = await pool.balanceOf(sender);
+
+              await expect(pool.joinGivenOut({ from: sender, bptOut, token: poolTokens.get(0) })).to.be.revertedWith(
+                'INVALID_JOIN_EXIT_WHILE_PAUSED'
+              );
+            });
+
+            it('allows proportionate joins', async () => {
+              const startingBpt = await pool.balanceOf(sender);
+
+              const { amountsIn } = await pool.multiJoinGivenOut({ from: sender, bptOut: startingBpt });
+
+              const endingBpt = await pool.balanceOf(sender);
+              expect(endingBpt).to.be.gt(startingBpt);
+              expect(amountsIn).to.deep.equal(initialBalances);
+            });
+
+            it('disallows disproportionate joins (multi-token, 95%)', async () => {
+              const amountsIn = [...initialBalances];
+              amountsIn[0] = fp(0.95);
+
+              await expect(pool.joinGivenIn({ from: sender, amountsIn })).to.be.revertedWith(
+                'INVALID_JOIN_EXIT_WHILE_PAUSED'
+              );
+            });;*/
```

Comment:
Let's not have commented code checked in - we'll add it once it's uncommented.

Reply:
Some of this is obsolete; rest will be uncommented (since we know have that join type) or deleted if unused